File: easy/calculate_distance.lua
for line in io.lines(arg[1]) do
  local x1, y1, x2, y2 = line:match("%((-?%d+), (-?%d+)%) %((-?%d+), (-?%d+)%)")
  local x, y = tonumber(x1) - tonumber(x2), tonumber(y1) - tonumber(y2)
  print(math.floor(math.sqrt(x * x + y * y)))
end

File: easy/calculate_distance.hs
import System.Environment (getArgs)

stripChars :: String -> String -> String
stripChars = filter . flip notElem

distance                 :: [Double] -> String
distance [x1, y1, x2, y2] = show . floor $ sqrt (x*x + y*y)
                          where x = x1 - x2
                                y = y1 - y2

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (distance . map read . words) . lines $ stripChars "()," input
